### PR TITLE
Shoulderfixes

### PR DESCRIPTION
**Pull request by YJ-7274**

### CODE REVIEW

**Review by prensing — CHANGES_REQUESTED** [review]
I suspect most of the class instance variables in Climber can be deleted.

**Comment by prensing** [comment]
This change must have been branched of plotTrajectory, which is not "right", but also not a problem.

File: src/main/java/frc/robot/commands/DeployIntake.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc.robot.Constants;
+import frc.robot.subsystems.Climber;
+
+public class DeployIntake extends CommandBase {
+    Climber climber;
+  //tests to see if we have started deploying the robot yet ;P
+    boolean started = false;
+    public DeployIntake(Climber climber){
+      this.climber = climber;
```
Comment (prensing) [inline src/main/java/frc/robot/commands/DeployIntake.java:18]: Please fix the indentation

File: src/main/java/frc/robot/commands/DeployIntake.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    }
+
+  // Called when the command is initially scheduled.
+  @Override
+  public void initialize() {
+    climber.shoulder.setVoltage(Constants.SHOULDER_SPEED_HOLD);
+    climber.shoulder.setIdleMode(IdleMode.kBrake);
```
Comment (prensing) [inline src/main/java/frc/robot/commands/DeployIntake.java:25]: "started" should be set to false in initialize().

File: src/main/java/frc/robot/commands/DeployIntake.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+  @Override
+  public void execute() {
+    //raise the shoulder a little so that the intake can unhook at the start of the match
+    if(!started){
+      climber.shoulder.setVoltage(0.3);
+    }
+    
+    //set started to true so that we don't raise the shoulder again.
+    started = true;
```
Comment (prensing) [inline src/main/java/frc/robot/commands/DeployIntake.java:36]: This should be inside the above if() block. No need to set it every time.

File: src/main/java/frc/robot/commands/DeployIntake.java [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+  public void initialize() {
+    climber.shoulder.setVoltage(Constants.SHOULDER_SPEED_HOLD);
+    climber.shoulder.setIdleMode(IdleMode.kBrake);
+  }
+
+  @Override
+  public void execute() {
+    //raise the shoulder a little so that the intake can unhook at the start of the match
+    if(!started){
+      climber.shoulder.setVoltage(0.3);
```
Comment (prensing) [inline src/main/java/frc/robot/commands/DeployIntake.java:32]: This is the "hold" voltage, which is not what you want. We need a voltage to raise the shoulder just a little.
Make sure to create a "final static" above to hold the value, and give it a good name.

File: src/main/java/frc/robot/commands/DeployIntake.java [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+    }
+    
+    //set started to true so that we don't raise the shoulder again.
+    started = true;
+
+    //slowly lower the shoulder
+    if(!climber.shoulderAtMinHeight()) {
+      climber.shoulder.setVoltage(-0.1);
```
Comment (prensing) [inline src/main/java/frc/robot/commands/DeployIntake.java:40]: The "-0.1" should be put in the class as a "static final" with a good name.

File: src/main/java/frc/robot/commands/DeployIntake.java [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+  //tests to see if we have started deploying the robot yet ;P
+    boolean started = false;
+    public DeployIntake(Climber climber){
+      this.climber = climber;
+    }
+
+  // Called when the command is initially scheduled.
+  @Override
+  public void initialize() {
+    climber.shoulder.setVoltage(Constants.SHOULDER_SPEED_HOLD);
```
Comment (prensing) [inline src/main/java/frc/robot/commands/DeployIntake.java:24]: This is no longer needed. Was just for testing.

File: src/main/java/frc/robot/commands/DeployIntake.java (hunk 5 above)
Comment (prensing) [reply src/main/java/frc/robot/commands/DeployIntake.java:40]: The first loop will not actually work. You will set the voltage (currently 0.3), set start, and then immediately set voltage to -0.1.

**Review by prensing — CHANGES_REQUESTED** [review]

**Comment by YJ-7274** [comment]
I addressed those changes ;)

**Comment by prensing** [comment]
YJ the code is looking decent, but somehow along the way, a bunch of changes got lost. The code from the shed was this:
https://github.com/ligerbots/InfiniteRecharge2021/commit/0be780148a468fd734e2d65dca9325334a46f1bd#diff-1000d31648c1ff8b42b5deec3bee8047af990c42a3ff199d5ed9423a8046aaf3

There were a few changes that were undone:
* Climber.shoulderBelowHeight was removed
* there are 3 voltages need: 
    - raise shoulder a little  (you have 0.5;  good starting guess)
    - lower shoulder quickly  (-3.0)
    - near bottom, lower shoulder slowly (-0.3)

**Review by prensing — APPROVED** [review]

**Comment by prensing** [comment]
Might need tuning of the initial raise of the shoulder, but good for now.

**Pull request merged**